akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -272,12 +274,6 @@ private[stream] object TcpConnectionStage {
 
       override def onUpstreamFailure(ex: Throwable): Unit = {
         if (connection != null) {
-          if (interpreter.log.isDebugEnabled) {
```

Comment:
> Does it have to include the stack trace though?

Yes, that is the whole point. You made a mistake and an exception bubbled up here, tearing down the connection. Of course you want to know where the exception was thrown. I am not reverting this so then other people start to complain why they don't get proper information. This is a very strong no from me.